src/utils/file.utils.ts

```diff
@@ -0,0 +1,194 @@
+ * @returns {Promise<Blob>} A Promise that resolves with the compressed image as a Blob.
+ */
+export const compressImage = ({ src, filename, options }: TCompressImage): Promise<Blob> => {
+    const { maxWidth = DEFAULT_IMAGE_WIDTH, quality = DEFAULT_IMAGE_QUALITY } = options || {};
+
+    return new Promise((resolve, reject) => {
+        const image = new Image();
+        image.src = src;
+        image.onload = () => {
+            const canvas = document.createElement("canvas");
+            const canvas_context = canvas.getContext("2d");
+            if (!canvas_context || !(canvas_context instanceof CanvasRenderingContext2D)) {
+                return reject(new Error("Failed to get 2D context"));
+            }
+
+            if (image.naturalWidth > maxWidth) {
+                const width = DEFAULT_IMAGE_WIDTH;
+                const scaleFactor = width / image.naturalWidth;
+                canvas.width = width;
+                canvas.height = image.naturalHeight * scaleFactor;
+            } else {
+                canvas.width = image.naturalWidth;
+                canvas.height = image.naturalHeight;
+            }
+
+            canvas_context.fillStyle = "transparent";
+            canvas_context.fillRect(0, 0, canvas.width, canvas.height);
+            canvas_context.save();
+            canvas_context.drawImage(image, 0, 0, canvas.width, canvas.height);
+
+            canvas.toBlob(
+                (blob) => {
+                    if (!blob) return;
+                    const modified_filename = filename.replace(/\.[^/.]+$/, ".jpg");
+                    const file = new Blob([blob], { type: "image/jpeg" });
+                    file.lastModifiedDate = Date.now();
+                    file.name = modified_filename;
+                    resolve(file);
+                },
+                "image/jpeg",
+                quality,
+            );
+        };
+    });
+};
+
+/**
+ * Convert a File to a Base64 encoded image representation.
+ * @param {File} file - The File object to convert to Base64.
+ * @returns {Promise<TBase64Image>} A Promise that resolves with an object containing the Base64 image data and the filename.
+ */
+export const convertToBase64 = (file: File): Promise<TBase64Image> => {
+    return new Promise((resolve) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onloadend = () => {
+            resolve({
+                src: reader.result?.toString() || "",
+                filename: file.name,
+            });
+        };
+    });
+};
+
+/**
+ * Check if a given filename has a supported image format extension.
+ *
+ * @param {string} filename - The filename to check for a supported image format.
+ * @returns {boolean} True if the filename has a supported image format extension, false otherwise.
+ */
+export const isSupportedImageFormat = (filename: string) => /\.(png|jpg|jpeg|gif|pdf)$/gi.test(filename ?? "");
```

Comment:
I feel like this should be an option where the user can specify the formats themselves or we provide a default

Reply:
@yashim-deriv 
not fully understand your suggestion here.
the idea behind this function is to say does the file is an image or not based on the file extension.
not sure why we need to pass a format along with the filename, moreover, to pass the default value, what is the benefit to pass it?
if we want to know is particular format supports, in that case we should add isPNGSupported, isJPEGSupported, etc. 🤔